src/components/SecondaryPanes/Expressions.js

```diff
@@ -174,18 +178,30 @@ class Expressions extends PureComponent<Props, State> {
         return;
       }
 
+      e.persist();
       e.stopPropagation();
-      e.target.value = "";
-      this.props.addExpression(value);
+
+      const error = await parser.hasSyntaxError(value);
+      this.setState({ error });
+
+      if (error) {
+        e.target.value = value;
+        e.target.style.color = "red";
```

Comment:
we will want to style it w/ `CSS` so set a class when we're rendering the input that it has an error

Reply:
fixed